pyccel/codegen/printing/ccode.py

```diff
@@ -953,8 +953,11 @@ def _print_FunctionDef(self, expr):
             self._additional_args.append(expr.results)
         body  = self._print(expr.body)
         decs  = [Declare(i.dtype, i) if isinstance(i, Variable) else FuncAddressDeclare(i) for i in expr.local_vars]
-        if len(expr.results) <= 1 :
-            decs += [Declare(i.dtype, i) if isinstance(i, Variable) else FuncAddressDeclare(i) for i in expr.results]
+        if len(expr.results) == 1 :
+            if isinstance(expr.results[0], Variable):
+                decs += [Declare(expr.results[0].dtype, expr.results[0])]
```

Comment:
in return, you have property stmt, maybe you can use is here in the condition 
```PYTHON
if isinstance(expr.results[0], Variable) and not expr.results.stmt: # here if stmt is empty it can mean that this is not a tmp return . 
```
can you try this ?

Reply:
my bad don't do this :)